client/src/components/R&R/NewReview.jsx

```diff
@@ -4,72 +4,279 @@ import $ from 'jquery';
+          submitted: true,
+        });
+      });
+    }
+  }
+
+
+  reviewConstructor(data) {
+    const reqBody = {
+      product_id: this.props.pid,
+      rating: Number(this.state.stars),
+      summary: this.state.summary,
+      body: this.state.body,
+      recommend: Boolean(this.state.recommend),
+      name: this.state.nickname,
+      email: this.state.email,
+      characteristics: this.state.characteristics
+    }
+    console.log('reqbody >>>>', reqBody);
+    return reqBody;
+  }
+
+  dataValidator() {
+    console.log('current state >>>>', this.state);
+    let { stars, recommend, characteristics, summary, body, nickname, email, images } = this.state;
+    let errors = {};
 
+    if (stars === null) { errors.rating = ('Please select Overall Rating.'); }
+    if (Object.keys(characteristics).length === 0) { errors.characteristics = ('Please rate product characteristics.'); }
+    if (nickname.length === 0) { errors.nickname = ('Please provide your nickname.'); }
+    if (email.length === 0) { errors.email = ('Please provide your email.'); } // email format, images
+    if (email.indexOf('@') === -1) { errors.email = ('Please provide a valid email address.'); }
+    if (body.length < 50) { errors.body = ('Please make sure the Review Body is at least 50 characters.'); }
+
+    return errors;
+  }
+
+  toggleImgWindow(e) {
+    e.preventDefault(); // prevent reload of page
+    this.setState({
+      imgWindow: !this.state.imgWindow,
+    }, () => console.log('submit IMG button clicked, current state >>>>', this.state.imgWindow));
   }
 
   toggleModal() {
     this.setState({
-      show: !this.state.show
+      show: !this.state.show,
+      imgWindow: false,
+      errors: {},
+      showError: false
     })
   }
 
-  // stars, loading more questions/reviews, adding a question/review,
-  // styled-components
-  // All reviews will be saved per product.  Specific styles will not be accounted for within the review module.
   // 39333 to 40343
   render () {
+    let ratingArr = ['Poor', 'Fair', 'Average', 'Good', 'Great'];
+    let charNames = this.props.characteristics && Object.keys(this.props.characteristics);
+
     if (this.state.show) {
       return (
-        <div>
         <Background>
           <ModalWrapper>
-            <h2>Write Your Review</h2>
-            <h3>About the {this.props.name}</h3>
-        Submission form line 1         Submission form line 1        Submission form line 1         Submission form line 1        Submission form line 1
-        <br />
-        Submission form line 2         Submission form line 2        Submission form line 2         Submission form line 2        Submission form line 2
-        <br />
-        <div style={{display: 'inline'}}><Button style={{cursor: 'pointer'}} onClick={this.toggleModal}>Close</Button>&nbsp; &nbsp; &nbsp;<Button style={{cursor: 'pointer'}} onClick={this.submitReview}>Submit Review</Button></div>
-        </ModalWrapper>
+            <ExitButton>
```

Comment:
this is a big render, consider starting with a ternary that renders the shorter render block first, then follow with the longer one. i can see this longer render block being refactored out to a few functional components that mostly contain static code